src/instantsearch/instantsearch.ts

```diff
@@ -262,12 +264,12 @@ export class NgAisInstantSearch implements AfterViewInit, OnInit, OnDestroy {
     // custom algolia client agent
     if (!config.searchClient && !config.createAlgoliaClient) {
```

Comment:
As explained in private, this check doesn't make sense anymore because we now recommend using the `searchClient` with our official JS client (instead of using `appId` and `apiKey`).

We should change this check to something like:

```js
config.searchClient = config.searchClient || algoliasearch(config.appId, config.apiKey);

if (typeof config.searchClient.addAlgoliaAgent === 'function') {
  // add user agents
}
```

`createAlgoliaClient` is deprecated and should be removed in a next major btw (it's replaced by `searchClient`). So are `appId` and `apiKey`. We can add a comment to mention that.